client/lxd_containers.go

```diff
@@ -88,13 +89,37 @@ func (r *ProtocolLXD) CreateContainerFromBackup(args ContainerBackupArgs) (Opera
 		return nil, fmt.Errorf("The server is missing the required \"container_backup\" API extension")
 	}
 
-	// Send the request
-	op, _, err := r.queryOperation("POST", "/containers", args.BackupFile, "")
+	var err error
+	var op Operation
+
+	if args.PoolName == "" {
+		// Send the request
+		op, _, err = r.queryOperation("POST", "/containers", args.BackupFile, "")
+		if err != nil {
+			return nil, err
+		}
+
+		return op, nil
+	}
+
+	if !r.HasExtension("container_backup_override_pool") {
+		return nil, fmt.Errorf("The server is missing the required \"container_backup_override_pool\" API extension")
+	}
+
+	data, err := ioutil.ReadAll(args.BackupFile)
```

Comment:
That's a terrible idea, this may end up loading several GB of stuff into memory, possibly several time as part of the http query, we really can't do that (and have an open bug to track down an internal case of this causing LXD to take up to 16GB of RAM while importing a backup).